src/ucp/core/ucp_worker.c

```diff
@@ -1837,29 +1838,24 @@ ucp_worker_get_ep_config(ucp_worker_h worker, const ucp_ep_config_key_t *key,
     ++worker->ep_config_count;
 
     if (context->config.ext.proto_enable) {
-        if (context->config.features & UCP_FEATURE_TAG) {
-            /* Set threshold for short send */
-            ucp_proto_select_short_init(worker, &ep_config->proto_select,
-                                        ep_cfg_index, UCP_WORKER_CFG_INDEX_NULL,
-                                        UCP_OP_ID_TAG_SEND, 0,
-                                        ucp_ep_config_key_has_tag_lane(key) ?
-                                                UCP_PROTO_FLAG_TAG_SHORT :
-                                                UCP_PROTO_FLAG_AM_SHORT,
-                                        &tag_short);
-            /* short protocol should be either disabled, or use key->am_lane */
-            ucs_assert((tag_short.max_length_host_mem < 0) ||
-                       (tag_short.lane == key->am_lane));
+        if (ucp_ep_config_key_has_tag_lane(key)) {
```

Comment:
need to check UCP_FEATURE_TAG ?

Reply:
i think no, tag lane will not be initialized if no TAG feature is requested
 and there is check for supported feature inside `ucp_ep_config_short_init`